### DISCUSSION TITLE
Running cppcheck from docker in CLion?

### DISCUSSION THREAD

**Discussion by ukhan717** [johnthagen/clion-cppcheck#86] [Ideas]
Currently, I am using cppcheck on Windows with CLion and it is working properly.

![image](https://user-images.githubusercontent.com/16636313/232497902-e9502aaf-08c0-4dba-928e-a16284b6bbde.png)

However, I am interested in exploring the possibility of using cppcheck from Docker with CLion. I have installed cppcheck on Docker, since I am using the Docker toolchain, and it would be beneficial if CLion could utilize cppcheck directly from Docker.

Would it make sense to extend the scope of this plugin in that direction?
In general, it would be greatly appreciated if you could provide any suggestions to assist us in our pursuit.

Looking forward to your response,

Kind Regards,

**Comment by firewave** [comment]
Thanks for the suggestion.

I don't see that happening though since there are various issues:
- requires configuration/access to the Docker instance to run the command
- requires the Docker instance to have access to the file to analyze (plus all includes)
  - requires knowledge about the Docker mappings
- requires information to manage the process running inside Docker
- etc.

The plugin is tightly tied to the system of IDE installation (which in turn probably causes issue with shared configurations).

A more generic way to implement this would be via SSH but it basically has the same issues.

It is possible that already configured instances in a JetBrains IDE are available to plugins but I don't see us implement that at all.

The whole framework of the plugin is extremely barebones as it is, Heck, the whole plugin is basically a glorified wrapper around an XML. 😃 And it turns out the plugin is even incorrectly implemented (see #65), so it is amazing it works as well as it does.
